File: app/src/main/java/com/example/moviesearcher/model/repositories/MovieListRepository.java
package com.example.moviesearcher.model.repositories;

import android.util.Log;

import com.example.moviesearcher.model.data.Movie;
import com.example.moviesearcher.model.handlers.responses.MovieListAsyncResponse;
import com.example.moviesearcher.util.MovieDbUtil;
import com.example.moviesearcher.util.UrlUtil;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class MovieListRepository {

    public void fetchData(JSONObject objectResponse, HashMap<Integer, String> genres, MovieListAsyncResponse callback){
        Thread movieListThread = new Thread(() -> {
            List<Movie> movieList = new ArrayList<>();
            try {
                JSONArray jsonArray = objectResponse.getJSONArray(MovieDbUtil.KEY_RESULT_ARRAY);
                for (int i = 0; i < jsonArray.length(); i++) {
                    JSONObject jsonObject = jsonArray.getJSONObject(i);
                    Movie movie = new Movie();

                    movie.setPosterImageUrl(UrlUtil.getImageUrl(jsonObject.getString(MovieDbUtil.KEY_POSTER_PATH)));
                    movie.setId(jsonObject.getInt(MovieDbUtil.KEY_ID));
                    movie.setTitle(jsonObject.getString(MovieDbUtil.KEY_MOVIE_TITLE));
                    movie.setReleaseDate(jsonObject.getString(MovieDbUtil.KEY_RELEASE_DATE));
                    movie.setScore(jsonObject.getString(MovieDbUtil.KEY_MOVIE_SCORE));

                    JSONArray genresIds = jsonObject.getJSONArray(MovieDbUtil.KEY_MOVIE_GENRE_IDS_ARRAY);
                    List<String> genresList = new ArrayList<>();
                    for (int j = 0; j < genresIds.length(); j++) {
                        genresList.add(genres.get(genresIds.getInt(j)));
                    }
                    movie.setGenres(genresList);
                    movieList.add(movie);
                }
            } catch (JSONException e) {
                Log.d("JSONArrayRequest", "getMovieList: EXCEPTION OCCURRED");
            }
            if (callback != null) callback.processFinished(movieList);
        });
        movieListThread.setPriority(6);
        movieListThread.start();
    }
}

File: app/src/main/java/com/example/moviesearcher/model/repositories/MovieListRepository.kt
package com.example.moviesearcher.model.repositories

import android.util.Log
import com.example.moviesearcher.model.data.Movie
import com.example.moviesearcher.model.services.responses.MovieListAsyncResponse
import com.example.moviesearcher.util.*
import org.json.JSONException
import org.json.JSONObject
import java.util.*

class MovieListRepository() {

    fun fetchData(objectResponse: JSONObject, genres: HashMap<Int?, String?>, callback: MovieListAsyncResponse?) {
        val movieListThread = Thread(Runnable {
            val movieList: MutableList<Movie> = ArrayList()
            try {
                val jsonArray = objectResponse.getJSONArray(KEY_RESULT_ARRAY)
                for (i in 0 until jsonArray.length()) {
                    val jsonObject = jsonArray.getJSONObject(i)
                    val movie = Movie()
                    movie.posterImageUrl = getImageUrl(jsonObject.getString(KEY_POSTER_PATH))
                    movie.id = jsonObject.getInt(KEY_ID)
                    movie.title = jsonObject.getString(KEY_MOVIE_TITLE)
                    movie.releaseDate = jsonObject.getString(KEY_RELEASE_DATE)
                    movie.score = jsonObject.getString(KEY_MOVIE_SCORE)
                    val genresIds = jsonObject.getJSONArray(KEY_MOVIE_GENRE_IDS_ARRAY)
                    val genresList: MutableList<String?> = ArrayList()
                    for (j in 0 until genresIds.length()) {
                        genresList.add(genres[genresIds.getInt(j)])
                    }
                    movie.setGenres(genresList)
                    movieList.add(movie)
                }
            } catch (e: JSONException) {
                Log.d("JSONArrayRequest", "getMovieList: EXCEPTION OCCURRED")
            }
            callback?.processFinished(movieList)
        })
        movieListThread.priority = 6
        movieListThread.start()
    }
}